sound/soc/sof/ipc4.c

```diff
@@ -653,9 +653,24 @@ static int sof_ipc4_ctx_save(struct snd_sof_dev *sdev)
 	return sof_ipc4_set_core_state(sdev, SOF_DSP_PRIMARY_CORE, false);
 }
 
+static int sof_ipc4_set_pm_state(struct snd_sof_dev *sdev, u32 flags)
+{
+	struct sof_ipc4_msg msg;
+
+	memset(&msg, 0, sizeof(msg));
+
+	msg.primary = SOF_IPC4_MSG_TYPE_SET(SOF_IPC4_MOD_SET_D0IX);
+	msg.primary |= SOF_IPC4_MSG_DIR(SOF_IPC4_MSG_REQUEST);
+	msg.primary |= SOF_IPC4_MSG_TARGET(SOF_IPC4_MODULE_MSG);
+	msg.extension = flags;
+
+	return sof_ipc4_tx_msg(sdev, &msg, msg.data_size, NULL, 0, true);
```

Comment:
whre is this msg.data_size set? if not set why not use 0 here?

Reply:
msg is clear to 0.  will use 0 here . Thanks!